### ISSUE TITLE
problematic side-effects in test functions

### ISSUE THREAD

**Issue by brisvag** [napari/napari#5404]
## 🐛 Bug

While working on #4894, I noticed [some odd failures in tests](https://github.com/napari/napari/pull/4894#issuecomment-1346620948) which behaved differently depending on which tests were being run. Turns out, the issue is on `main` too, it was just changed slightly in that PR which exposed the issue. Get ready for some weird stuff :)

If you run `pytest napari/_tests/` (or the full test suite) on `main`, everything looks fine. However,

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods"
```

will fail on image and shapes. Later, I found out that this:

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods or test_mgui_forward_refs"
```

will instead fail on points, labels and shapes, and succeed on the rest. If you run any other function in the same file, however, we keep the original failure:

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods or test_magicgui_get_viewer"
```

This suggests that `test_mgui_forward_refs` has some side effects which change the behaviour of subsequent tests. However, whatever it does, it's not enough to get to the "all tests pass" behaviour we get from the full test suite. So I locked in the above tests, and looked for another function with side effects.

Long story short, the functions inside `test_mouse_bindings.py` need to run *before* the magicgui one. So if you run:

```bash
pytest napari/_tests/ -k "test_expected_number_of_layer_methods or test_mgui_forward_ref or _mouse_bindings"
```

It will fail, but if you run:

```bash
pytest napari/_tests/{test_mouse_bindings.py,test_magicgui.py,test_viewer.py}
```

You finally get the "successful" behaviour. Note that between the two last commands, only the order of execuction changed.

---

Any suggestions on how to fix this? It seems likely that monkeypatches and `Mock` are misbehaving.

cc @andy-sweet @Czaki

**Comment by Czaki** [comment]
Maybe not all imports are performed, so not every keybind is registered?

**Comment by brisvag** [comment]
Hopefully solved with #5406!
